src/hooks/useDelayToggleButtonState.js

```diff
@@ -0,0 +1,39 @@
+import {useState, useEffect, useRef} from 'react';
+
+/**
+ * Hook that manages a delayed toggle button state.
+ *
+ * @returns {Object} - An object containing isDelayButtonStateComplete and toggleDelayButtonState.
+ */
+export default function useDelayToggleButtonState() {
```

Comment:
The hook seems a little complicated than it should, can't we write it as follow:

```js
export default function useDelayToggleButtonState() {
    const [isDelayButtonStateComplete, setIsDelayButtonStateComplete] = useState(false);
    const toggleDelayButtonState = useCallback(() => setIsDelayButtonStateComplete(true), []);

    useEffect(() => {
        if (!isDelayButtonStateComplete) {
            return;
        }

        const timer = setTimeout(() => {
            setIsDelayButtonStateComplete(false);
        }, 1000);
        return () => clearTimeout(timer);
    }, [isDelayButtonStateComplete]);

    return {isDelayButtonStateComplete, toggleDelayButtonState};
}
```